plugins/events/create.js

```diff
@@ -220,15 +220,24 @@ module.exports = () => ({
 
             payload.sha = sha;
 
-            // If there is parentEvent, pass workflowGraph and sha to payload
+            // If there is parentEvent, pass workflowGraph, meta and sha to payload
             // Skip PR, for PR builds, we should always start from latest commit
             if (payload.parentEventId) {
                 const parentEvent = await eventFactory.get(parentEventId);
+                let mergedParameters = {};
 
                 payload.baseBranch = parentEvent.baseBranch || null;
 
                 if (!payload.meta.parameters && parentEvent.meta && parentEvent.meta.parameters) {
-                    payload.meta.parameters = parentEvent.meta.parameters;
+                    mergedParameters = parentEvent.meta.parameters;
+                }
+
+                if (Object.keys(payload.meta).length === 0) {
+                    payload.meta = { ...parentEvent.meta };
+                }
+
+                if (Object.keys(mergedParameters).length > 0) {
+                    payload.meta.parameters = mergedParameters;
                 }
```

Comment:
what about make this more simplified?
```suggestion
                // Merge parameters if they exist in the parent event and not in the payload
                if (!payload.meta.parameters && parentEvent.meta && parentEvent.meta.parameters) {
                    payload.meta.parameters = parentEvent.meta.parameters;
                }
                
                // Copy meta from parent event if payload.meta is empty
                if (Object.keys(payload.meta).length === 0) {
                    payload.meta = { ...parentEvent.meta };
                }
```

Reply:
If payload(RestartedEevnt) only has parameters in meta, I think your suggested method will not fetch parentEvent meta (and neither will my method).
I will have fixed this bug.